gtars/src/overlap/ailist.rs

```diff
@@ -188,20 +241,114 @@ where
         self.starts.is_empty()
     }
 }
+
+/// Find Iterator
+#[derive(Debug)]
+pub struct IterFind<'a, I, T>
+where
+    T: Eq + Clone + Send + Sync + 'a,
+    I: PrimInt + Unsigned + Clone + Send + Sync,
+{
+    inner: &'a AiList<I, T>,
+    header_list_idx: usize,
+    list_idx: Option<usize>,
+    start: I,
+    stop: I,
+}
+
+impl<'a, I, T> IterFind<'a, I, T>
+where
+    I: PrimInt + Unsigned + Clone + Send + Sync + 'a,
+    T: Eq + Clone + Send + Sync,
+{
+    fn new(ailist: &'a AiList<I, T>, start: I, stop: I) -> Self {
+        Self {
+            inner: ailist,
+            header_list_idx: 0,
+            list_idx: None,
+            start,
+            stop,
+        }
+    }
+}
+
+impl<'a, I, T> Iterator for IterFind<'a, I, T>
```

Comment:
Amazing thank you. I tried to do this a few weeks ago and couldn't get it. Appreacite this impl